src/test/test-fileio.c

```diff
@@ -624,11 +624,12 @@ static void test_fgetc(void) {
                 assert_se(safe_fgetc(f, &c) == 1);
                 assert_se(c == chars[i]);
 
-                assert_se(ungetc(c, f) != EOF);
-                assert_se(safe_fgetc(f, &c) == 1);
+                /* EOF is -1, and hence we can't push value 255 in this way */
+                assert_se(ungetc(c, f) != EOF || c == EOF);
+                assert_se(c == EOF || safe_fgetc(f, &c) == 1);
```

Comment:
A static analyzer I've been playing with for the last two days seems to be complaining (correctly) that [EOF should not be compared with a value of the 'char' type](https://evverx.github.io/pvs-studio-reports/sources/test-fileio.c_307.html#ln628).

By the way, as far as I know, it's still possible to [get a free licence for open source projects](https://www.viva64.com/en/b/0600/). In theory, the tool could be used to analyze the code when PRs are opened, but it's necessary to fix/suppress all the false positive first so as not to bother contributors. In case anyone interested, a full report can be found at https://evverx.github.io/pvs-studio-reports/. I particularly like an out-of-the-box "alloca in a loop" rule: https://evverx.github.io/pvs-studio-reports/sources/bus-objects.c_38.html#ln2117. @yuwata @poettering @keszybz @filbranden what do you think? Would it make sense to try to integrate it somehow?

Reply:
I suppressed 718 false positives with three comments. Now it should be a little bit easier to stumble upon real bugs like https://evverx.github.io/pvs-studio-reports/sources/networkctl.c_171.html#ln1054 :-)

Reply:
Having played with it a little bit more, I'd say that a more realistic option would be to analyze the code once a day, publish reports somewhere and share links to them in the hope than someone is going to fix all the bugs and false positives :-) I already have a licence so basically all I need to do is to create a repository where the reports would be stored, set up GitHub Pages there, create a bot with access to that repository, add a few lines to the Coverity stage and probably rename it to "Static Analysis" or something like that. That's absolutely doable and should help to see the tool in action. I could always turn it off if something goes wrong :-) Just let me know what you think.